tests/test_models/test_simple_models.py

```diff
@@ -729,3 +729,63 @@ def test_deadline_model_forecast_correct_with_big_horizons(two_month_ts):
 )
 def test_save_load(model, example_tsds):
     assert_model_equals_loaded_original(model=model, ts=example_tsds, transforms=[], horizon=3)
+
+
+@pytest.mark.parametrize("method_name", ("forecast", "predict"))
+@pytest.mark.parametrize(
+    "window, seasonality, expected_components_names",
+    ((1, 7, ["target_component_lag_7"]), (2, 7, ["target_component_lag_7", "target_component_lag_14"])),
+)
+def test_sma_model_predict_components_correct_names(
+    example_tsds, method_name, window, seasonality, expected_components_names, horizon=10
+):
+    model = SeasonalMovingAverageModel(window=window, seasonality=seasonality)
+    model.fit(example_tsds)
+    to_call = getattr(model, method_name)
+    forecast = to_call(ts=example_tsds, prediction_size=horizon, return_components=True)
+    assert sorted(forecast.target_components_names) == sorted(expected_components_names)
+
+
+@pytest.mark.parametrize("method_name", ("forecast", "predict"))
+@pytest.mark.parametrize("window", (1, 3, 5))
+@pytest.mark.parametrize("seasonality", (1, 7, 14))
+def test_sma_model_predict_components_sum_up_to_target(example_tsds, method_name, window, seasonality, horizon=10):
+    model = SeasonalMovingAverageModel(window=window, seasonality=seasonality)
+    model.fit(example_tsds)
+    to_call = getattr(model, method_name)
+    forecast = to_call(ts=example_tsds, prediction_size=horizon, return_components=True)
+
+    target = forecast.to_pandas(features=["target"])
+    target_components_df = forecast.get_target_components()
+    np.testing.assert_array_almost_equal(
+        target.values, target_components_df.sum(axis=1, level="segment").values, decimal=10
+    )
+
+
+@pytest.fixture
+def simple_ts() -> TSDataset:
+    periods = 100
+    timestamp = pd.date_range("2020-01-01", periods=periods)
+    df1 = pd.DataFrame({"timestamp": timestamp, "segment": "segment_1", "target": np.arange(1, periods + 1)})
+    df2 = pd.DataFrame({"timestamp": timestamp, "segment": "segment_2", "target": 100 + np.arange(1, periods + 1)})
+    df = pd.concat([df1, df2]).reset_index(drop=True)
+    df = TSDataset.to_dataset(df)
+    tsds = TSDataset(df, freq="D")
+
+    return tsds
+
+
+@pytest.mark.parametrize(
+    "method_name, expected_values",
+    (("forecast", [[96, 196], [97, 197], [96, 196]]), ("predict", [[96, 196], [97, 197], [98, 198]])),
+)
+def test_sma_model_predict_components_correct(
```

Comment:
Seems like this test is redundant as we tested the number of components and consistency with the predictions.

Reply:
Here we check that components themself are correct(because here we can drive them explicitly)